### PR TITLE
Extract serve.rs into a separate crate

### PR DESCRIPTION
**Pull request by tirr-c**
## Description
* Extracted `serve.rs` into a separate crate.
* Added public method `App::into_http_service`.
* Newly added default feature `hyper` will determine whether `App::serve` will be present or not.
* Added an integration test that utilizes a mock backend.

## Motivation and Context
Part of #53. Along with #139, this will help make Tide more modular. A mock backend and an integration test are added to demonstrate how backends can be switched.

## How Has This Been Tested?
Wrote a simple integration test and it passed.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rust-net-web/tide/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: http-service-hyper/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+authors = [
+    "Aaron Turon <aturon@mozilla.com>",
```
Comment (yoshuawuyts) [inline http-service-hyper/Cargo.toml]: @tirr-c you should add yourself here; if I'm included, then you definitely should be!

**Comment by tirr-c** [comment]
@yoshuawuyts I agree on moving `http-service-hyper` into another repository. `http-service` might have the crate as a workspace member, I suppose? Or do you think a dedicated repository would be better?

**Comment by aturon** [comment]
@tirr-c Putting it in the `http-service` repo makes sense to me! Want to make a PR there?

**Comment by tirr-c** [comment]
@aturon I opened a PR to http-service, please check it out!

**Comment by aturon** [comment]
@tirr-c merged and published! Feel free to update this PR  to use the external crate now 👍

**Comment by tirr-c** [comment]
@aturon @yoshuawuyts I've updated PR to use the external crate version.

**Pull request merged**